cryptography/src/bls12381/mod.rs

```diff
@@ -6,6 +6,10 @@
 //!
 //! - `portable`: Enables `portable` feature on `blst` (<https://github.com/supranational/blst?tab=readme-ov-file#platform-and-language-compatibility>).
 
+#[cfg(not(feature = "std"))]
+extern crate alloc;
+
+#[cfg(feature = "std")]
```

Comment:
lots of uses of HashMap/HashSet within. Not sure how important this is to include in no_std?

Reply:
Most of the stuff we'd want to do in no_std is in `primitives` so having that in is huge.